// @bitiumagency/web3-react-modal/lib/cjs/lib/components/wallets-modal/index.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var useOutsiteClick_1 = __importDefault(require("../../../hooks/useOutsiteClick"));
var store_1 = __importDefault(require("../../../store/store"));
var providers_logo_1 = __importDefault(require("../providers-logo"));
var WalletbuttonModalItem = styled_components_1.default.button(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  width: 100%;\n  padding: 10px;\n  border: none;\n  border-radius: 5px;\n  background-color: #fafafa;\n  font-size: 14px;\n  color: #212121;\n  cursor: pointer;\n  margin-bottom: 10px;\n  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.1);\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  &:hover {\n    background-color: #f5f5f5;\n    box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.2);\n  }\n"], ["\n  width: 100%;\n  padding: 10px;\n  border: none;\n  border-radius: 5px;\n  background-color: #fafafa;\n  font-size: 14px;\n  color: #212121;\n  cursor: pointer;\n  margin-bottom: 10px;\n  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.1);\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  &:hover {\n    background-color: #f5f5f5;\n    box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.2);\n  }\n"])));
var WalletbuttonModalTitle = styled_components_1.default.span(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  font-size: 14px;\n  font-weight: bold;\n  color: #212121;\n  margin-bottom: 25px;\n  margin-top: 15px;\n"], ["\n  font-size: 14px;\n  font-weight: bold;\n  color: #212121;\n  margin-bottom: 25px;\n  margin-top: 15px;\n"])));
var WalletbuttonModalContainer = styled_components_1.default.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  overflow-y: auto;\n  width: 250px;\n  max-height: 70%;\n  position: absolute;\n  left: 50%;\n  top: 50%;\n  transform: translate(-50%, -50%);\n  padding: 10px;\n  margin: 0px;\n  border: none;\n  border-radius: 5px;\n  background-color: #fafafa;\n  font-size: 14px;\n  font-weight: bold;\n  color: #000;\n  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.3);\n  &:hover {\n    background-color: #f5f5f5;\n    box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.5);\n  }\n  @media (max-width: 768px) {\n    width: 80%;\n  }\n  &::-webkit-scrollbar {\n    width: 5px;\n    background: transparent;\n  }\n  &::-webkit-scrollbar-track {\n    background: rgba(255, 255, 255, 0.7);\n    border-radius: 5px;\n  }\n  &::-webkit-scrollbar-thumb {\n    background: rgba(0, 0, 0, 0.4);\n    border-radius: 5px;\n  }\n"], ["\n  overflow-y: auto;\n  width: 250px;\n  max-height: 70%;\n  position: absolute;\n  left: 50%;\n  top: 50%;\n  transform: translate(-50%, -50%);\n  padding: 10px;\n  margin: 0px;\n  border: none;\n  border-radius: 5px;\n  background-color: #fafafa;\n  font-size: 14px;\n  font-weight: bold;\n  color: #000;\n  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.3);\n  &:hover {\n    background-color: #f5f5f5;\n    box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.5);\n  }\n  @media (max-width: 768px) {\n    width: 80%;\n  }\n  &::-webkit-scrollbar {\n    width: 5px;\n    background: transparent;\n  }\n  &::-webkit-scrollbar-track {\n    background: rgba(255, 255, 255, 0.7);\n    border-radius: 5px;\n  }\n  &::-webkit-scrollbar-thumb {\n    background: rgba(0, 0, 0, 0.4);\n    border-radius: 5px;\n  }\n"])));
function WalletsModal() {
    var _this = this;
    var _a = (0, store_1.default)(), setModalIsOpen = _a.setModalIsOpen, modalIsOpen = _a.modalIsOpen, useWeb3React = _a.useWeb3React, activateInjected = _a.activateInjected, connectors = _a.connectors;
    var _b = useWeb3React(), activate = _b.activate, chainId = _b.chainId;
    var onSelect = function () { return setModalIsOpen(false); };
    var activeInjected = function () { return __awaiter(_this, void 0, void 0, function () {
        return __generator(this, function (_a) {
            activateInjected(activate, chainId);
            onSelect();
            return [2 /*return*/];
        });
    }); };
    var activeConnector = function (connector) { return __awaiter(_this, void 0, void 0, function () {
        return __generator(this, function (_a) {
            activate(connector);
            onSelect();
            return [2 /*return*/];
        });
    }); };
    var modalContainer = (0, react_1.useRef)(null);
    (0, useOutsiteClick_1.default)(modalContainer, function () {
        setModalIsOpen(false);
    });
    return modalIsOpen ? (react_1.default.createElement("div", { style: {
            width: "100%",
            height: "100%",
            position: "absolute",
            backgroundColor: "rgba(0,0,0,0.5)",
            zIndex: "1",
            left: 0,
            top: 0
        }, className: "walletbutton-modal-root" },
        react_1.default.createElement(WalletbuttonModalContainer, { ref: modalContainer, className: "walletbutton-modal-container" },
            react_1.default.createElement("div", { style: {
                    height: "100%",
                    width: "100%",
                    display: "flex",
                    flexDirection: "column",
                    justifyContent: "center",
                    alignItems: "center"
                } },
                react_1.default.createElement(WalletbuttonModalTitle, { className: "walletbutton-modal-title" }, "Select a wallet"),
                react_1.default.createElement(WalletbuttonModalItem, { className: "walletbutton-modal-item", onClick: activeInjected },
                    "Metamask",
                    react_1.default.createElement("div", { style: { width: "35px", height: "35px" } },
                        react_1.default.createElement(providers_logo_1.default, { provider: "injected" }))),
                connectors.map(function (connector) { return (react_1.default.createElement(WalletbuttonModalItem, { className: "walletbutton-modal-item", key: connector.id, onClick: function () { return activeConnector(connector.connector); } },
                    connector.title,
                    connector.logo ? (connector.logo) : (react_1.default.createElement("div", { style: { width: "35px", height: "35px" } },
                        react_1.default.createElement(providers_logo_1.default, { provider: connector.id }))))); }))))) : (react_1.default.createElement(react_1.default.Fragment, null));
}
exports.default = WalletsModal;
var templateObject_1, templateObject_2, templateObject_3;
